SPECS-EXTENDED/podman/podman.spec

```
@@ -25,190 +8,139 @@ Distribution:   Mariner
-# N.foo if released, 0.N.foo if unreleased
-# Rawhide almost always ships unreleased builds,
-# so release tag should be of the form 0.N.foo
-Release: 5%{?dist}
-Summary: Manage Pods, Containers and Container Images
-License: ASL 2.0
-URL: https://%{name}.io/
-Source0: %{download_url}#/%{name}-%{version}.tar.gz
-Source1: https://github.com/containers/dnsname/archive/c654c95366ac5f309ca3e5727c9b858864247328/dnsname-c654c95.tar.gz
-Provides: %{name}-manpages = %{version}-%{release}
-Obsoletes: %{name}-manpages < %{version}-%{release}
-# If go_compiler is not set to 1, there is no virtual provide. Use golang instead.
-BuildRequires: golang
-BuildRequires: glib2-devel
-BuildRequires: glibc-devel
-BuildRequires: glibc-static
-BuildRequires: git
-BuildRequires: go-md2man
-BuildRequires: go-rpm-macros
-BuildRequires: gpgme-devel
-BuildRequires: libassuan-devel
-BuildRequires: libgpg-error-devel
-BuildRequires: libseccomp-devel
-BuildRequires: libselinux-devel
-BuildRequires: pkgconfig
-BuildRequires: make
-BuildRequires: systemd
-BuildRequires: systemd-devel
-Requires: containernetworking-plugins >= 0.8.6-1
-Requires: iptables
-Requires: nftables
-Requires: conmon >= 2.0.18-1
-Obsoletes: oci-systemd-hook <= 0.2.0-3
-%if 0%{?fedora}
-BuildRequires: btrfs-progs-devel
-BuildRequires: ostree-devel
-Requires: containers-common >= 1.2.0-9
-Recommends: fuse-overlayfs >= 0.3-8
-Recommends: crun >= 0.15-5
-%else
-Requires: containers-common >= 1.2.0-9
-%endif
-%if 0%{?fedora} || 0%{?centos} >= 8
-Recommends: %{name}-plugins = %{version}-%{release}
-Recommends: catatonit
-Requires: (container-selinux if selinux-policy)
-Recommends: runc
-Recommends: slirp4netns >= 0.3.0-2
-Recommends: libvarlink-util
-%else
-Requires: %{name}-plugins = %{version}-%{release}
-Requires: catatonit
-Requires: container-selinux
-Requires: moby-runc
-Requires: slirp4netns >= 0.3.0-2
-%endif
+%global commit_plugins 18822f9a4fb35d1349eb256f4cd2bfd372474d84
+
+# gvproxy
+%global repo_gvproxy gvisor-tap-vsock
+# https://github.com/containers/gvisor-tap-vsock
+%global import_path_gvproxy %%{provider}.%{provider_tld}/%{project}/%{repo_gvproxy}
+%global git_gvproxy https://%{import_path_gvproxy}
+%global commit_gvproxy 4ee84d66bd86668f011733d8873989b5862bcd07
+
+%global built_tag v4.1.1
+
+Name:           podman
+Version:        4.1.1
+Release:        1%{?dist}
+License:        ASL 2.0 and BSD and ISC and MIT and MPLv2.0
+Summary:        Manage Pods, Containers and Container Images
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            https://%{name}.io/
+# Source0:      %{git0}/archive/%{built_tag}.tar.gz
+Source0:        %{name}-4.1.1.tar.gz
+# Source1:      %{git_plugins}/archive/%{commit_plugins}/%{repo_plugins}-%{commit_plugins}.tar.gz
+Source1:        dnsname-18822f9.tar.gz
+# Source2:      %{git_gvproxy}/archive/%{commit_gvproxy}/%{repo_gvproxy}-%{commit_gvproxy}.tar.gz
+Source2:        gvisor-tap-vsock-4ee84d6.tar.gz
```

Please use this notation, it works better with our tooling:
```suggestion
Source0:        %{git0}/archive/%{built_tag}.tar.gz#/%{name}-4.1.1.tar.gz
Source1:        %{git_plugins}/archive/%{commit_plugins}/%{repo_plugins}-%{commit_plugins}.tar.gz#/%{repo_plugins}-18822f9.tar.gz
Source2:        %{git_gvproxy}/archive/%{commit_gvproxy}/%{repo_gvproxy}-%{commit_gvproxy}.tar.gz#/%{repo_gvproxy}-4ee84d6.tar.gz
```

Please also bring back `shortcommit_plugins`, so we can use that instead of the hard-coded `18822f9` in `Source1`. This will make future updates easier and we'll know where these numbers came from.

Similarly, let's add `shortcommit_gvproxy` for `Source2`.

---

updated.

---

Added short commit macro in podman.